libparsec/crates/platform_mountpoint/src/unix/filesystem.rs

```diff
@@ -216,8 +217,90 @@ impl fuser::Filesystem for Filesystem {
+
+        // Note Fuser already set the following capabilities:
+        // - On Linux: FUSE_ASYNC_READ | FUSE_BIG_WRITES | FUSE_MAX_PAGES
+        // - On macOS: FUSE_ASYNC_READ | FUSE_CASE_INSENSITIVE | FUSE_VOL_RENAME | FUSE_XTIMES | FUSE_MAX_PAGES
+
+        // `FUSE_POSIX_LOCKS` (remote locking support) is not enabled since we don't
+        // implement `getlk()` & `setlk()` handlers.
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L179
+
+        // `FUSE_FLOCK_LOCKS` is not enabled since we don't implement `flock()` handler.
+        // Instead we rely on in-kernel POSIX lock emulation.
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L246
+
+        // `FUSE_ASYNC_DIO` is not enabled since we don't do direct I/O.
+        // See:
+        // - https://www.kernel.org/doc/html/latest/filesystems/fuse-io.html
+        // - https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L322
+
+        // Do not send separate SETATTR request before open(O_TRUNC).
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L188
+        config
+            .add_capabilities(fuser::consts::FUSE_ATOMIC_O_TRUNC)
+            .expect("Capability available");
+
+        // Support IOCTL on directories.
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L253
+        config
+            .add_capabilities(fuser::consts::FUSE_IOCTL_DIR)
+            .expect("Capability available");
+
+        // Tell FUSE the data may change without going through the filesystem, hence attributes
+        // validity should be checked on each access (leading to `getattr()` on timeout).
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L275
+        config
+            .add_capabilities(fuser::consts::FUSE_AUTO_INVAL_DATA)
+            .expect("Capability available");
+
+        // Enable `readdirplus` support (readdir + lookup in one a single go).
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L283
+        config
+            .add_capabilities(fuser::consts::FUSE_DO_READDIRPLUS)
+            .expect("Capability available");
+
+        // Adaptative readdirplus optimization support: still use readdir when lookup is
+        // has already been done by the previous readdirplus.
+        // See https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L311
+        config
+            .add_capabilities(fuser::consts::FUSE_READDIRPLUS_AUTO)
+            .expect("Capability available");
+
+        // TODO: `FUSE_SPLICE_READ` seems to require `write_buf()` to be implemented, which is
```

Comment:
Why read require `write_buf`, did you mean `read_buf` ?

Can you create an issue on `fuser` side ?

Reply:
Nop it's `write_buf` 

> Can you create an issue on fuser side ?

This is not related to fuser, it's [documented as such on libfuse](https://github.com/libfuse/libfuse/blob/2aeef499b84b596608181f9b48d589c4f8ffe24a/include/fuse_common.h#L227-L233)

I've created an issue on fuser side to try to better understand what is the support of `write_buf` on fuser: https://github.com/cberner/fuser/issues/298